### ISSUE TITLE
Potential memory leak

### ISSUE THREAD

**Issue by soonho-tri** [dreal/dreal4#51]
https://github.com/dreal/dreal4/blob/0a3b1c68351bc2f29cb4a0dec5e875894707ded9/dreal/util/ibex_converter.cc#L93-L99

Changing line 98 to `return &ibex::ExprConstant::new_scalar(c);` causes memory leaks. I'm investigating the issue.

**Comment by soonho-tri** [comment]
It turns out it's because of the optimization in IBEX. For example (from [ibex_ExprCtr.h](https://github.com/ibex-team/ibex-lib/blob/af48e38847414818913b6954e1b1b3050aa14593/src/symbolic/ibex_ExprCtr.h#L53-L55)):

```c++
inline const ExprCtr& ExprNode::operator=(const ExprNode& right) const {
  return *new ExprCtr(right.is_zero()? *this : *this-right, EQ);
}
```

Constructing `e1 = e2`, if `e2` is zero, then ibex does not include `e2` in the return value. As a result, one has to delete `e2` manually/explicitly. Similar issues are there in other operators (`<`, `<=`, ...).

**Comment by soonho-tri** [comment]
Fixed by 8fb371e1f5995820ee702ce34da4b7c11274d2e9
